operator/operator.php

```diff
@@ -70,6 +82,19 @@ public function set_user_categories($forum_id)
 		// TODO
 		// 1.if user is registered, update the db with serialized array of collapsed category data ($this->collapsed_categories)
 		// 2.set their cookie too by calling set_cookie_categories()
+
+		if ($this->user->data['is_registered'])
+		{
+			$sql = 'UPDATE ' . USERS_TABLE . '
+				SET collapsible_categories = ' . $this->db->sql_escape(serialize($this->collapsed_categories)) . '
+				WHERE user_id = ' . (int) $this->user->data['user_id'];
+			$this->db->sql_query($sql);
+		}
+
+		if (!$this->user->data['is_bot'])
```

Comment:
I would put this is_bot check at the top of the method and just return false.

Then this set_cookie_cats can be left outside of a conditional and it should return a value t/f, ie:
`return $this->set_cookie_categories($forum_id);`